app/src/main/java/org/simple/clinic/summary/PatientSummaryScreen.kt

```diff
@@ -194,6 +194,7 @@ class PatientSummaryScreen :
   }
 
   override fun uiRenderer(model: PatientSummaryModel): ViewRenderer<PatientSummaryModel> {
+    modelUpdateCallback?.invoke(model)
```

Comment:
Just FYI, this method is only invoked once when the Mobius loop is setup. This seems like something we want to be invoked for every model update, right?

Reply:
I have changed the `RxMobiusBridge` to trigger every time model changes happen in the connection. It seems to be working correctly 🤔

Reply:
This will end up creating a new instance of the UI renderer every time the model changes for all screens which seems a little wasteful. 

If what we want is for the `modelUpdateCallback` to be invoked every time the `PatientSummaryModel` is updated, we can instead change `PatientSummaryViewRenderer` to expose a `(PatientSummaryModel) -> Unit` constructor property which we can then invoke whenever the model changes and trigger the callback from there.

What do you think?

Reply:
So if I understood that correctly, we will trigger that constructor property in the `ViewRenderer#render` method right?

Reply:
Yup

Reply:
The Mobius concept for this is a `Model Update Listener`, it just so happens for us that until now, all our `Model Update Listener` types were view renderers. A better solution would be to be able to accept a list of model update listeners, which will generally be a list with one item, a `ViewRenderer`. In specialised cases like this, we could send more than one listener, a `ViewRenderer` and maybe something that can invoke a callback like we need here.

^ Can you add a refactoring ticket to track this and the Delegatable fragment I mentioned in a previous comment?

Reply:
Tracking here: https://app.clubhouse.io/simpledotorg/story/2647/accept-list-of-model-update-listeners-when-creating-screens-with-mobius